src/Interpreters/InterpreterShowSettingQuery.cpp

```diff
@@ -0,0 +1,37 @@
+#include <Interpreters/InterpreterShowSettingQuery.h>
+
+#include <Common/quoteString.h>
+#include <IO/Operators.h>
+#include <IO/WriteBufferFromString.h>
+#include <Parsers/ASTShowSettingQuery.h>
+#include <Parsers/formatAST.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/executeQuery.h>
+
+
+namespace DB
+{
+
+
+InterpreterShowSettingQuery::InterpreterShowSettingQuery(const ASTPtr & query_ptr_, ContextMutablePtr context_)
+    : WithMutableContext(context_)
+    , query_ptr(query_ptr_)
+{
+}
+
+
+String InterpreterShowSettingQuery::getRewrittenQuery()
+{
+    const auto & query = query_ptr->as<ASTShowSettingQuery &>();
+    return fmt::format(R"(SELECT value FROM system.settings WHERE name = '{0}')", query.getSettingName());
```

Comment:
Idea: MySQL's equivalent [statement](https://dev.mysql.com/doc/refman/8.0/en/show-variables.html) has `LIKE` filters. I think that would be useful here too. With that, consider to print not only the setting value but also the setting name (again similar to what MySQL does).

Reply:
We already have `SHOW SETTINGS LIKE|ILIKE name` query. We just did not have simple `SHOW SETTING`, analog of `SHOW` in PostgreSQL.